### PR TITLE
fix/peer_manager: remove approved candidates from map

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/peer_manager.rs [hunk 1]
```diff
@@ -698,10 +680,8 @@ impl PeerManager {
         }
 
         if have_candidate {
```
Comment (Viv-Rajkumar) [inline src/peer_manager.rs]: guess that got missed in the diff `if !have_candidate {`

Comment (Fraser999) [inline src/peer_manager.rs]: Well spotted.  I had to apply the diff manually - `git apply` failed.

File: src/peer_manager.rs [hunk 2]
```diff
@@ -629,26 +615,26 @@ impl PeerManager {
...
+                                                        target_size: target_size,
+                                                        difficulty: difficulty,
+                                                        seed: seed,
+                                                        proof: VecDeque::new(),
+                                                    });
+                (Ok(true), Some(state))
             }
+        } else if self.peer_map
+                      .get_by_name(pub_id.name())
+                      .map_or(false, |peer| peer.valid()) {
```
Comment (afck) [inline src/peer_manager.rs]: `.map_or(false, Peer::valid)`?

**Pull request merged**